force_bdss/core_evaluation_driver.py

```diff
@@ -31,18 +31,18 @@ def application_started(self):
         mco_bundle = mco_model.bundle
         mco_communicator = mco_bundle.create_communicator()
 
-        parameters = mco_communicator.receive_from_mco(mco_model)
+        mco_data_values = mco_communicator.receive_from_mco(mco_model)
 
         ds_results = []
         for ds_model in workflow.data_sources:
             ds_bundle = ds_model.bundle
             data_source = ds_bundle.create_data_source()
-            ds_results.append(data_source.run(ds_model, parameters))
+            ds_results.extend(data_source.run(ds_model, mco_data_values))
 
         kpi_results = []
         for kpic_model in workflow.kpi_calculators:
             kpic_bundle = kpic_model.bundle
             kpi_calculator = kpic_bundle.create_kpi_calculator()
-            kpi_results.append(kpi_calculator.run(kpic_model, ds_results))
+            kpi_results.extend(kpi_calculator.run(kpic_model, ds_results))
```

Comment:
You are not passing the DataValues coming from the MCO. Will this come later with the slots ?